packages/keystone/lib/ListTypes/list.js

```diff
@@ -833,9 +834,10 @@ module.exports = class List {
     const access = await this.checkListAccess(context, data, operation, extraData);
 
     const existingItems = await this.getAccessControlledItems(ids, access);
+    const existingItemsById = arrayToObject(existingItems, 'id');
 
     const itemsToUpdate = zipObj({
-      existingItem: existingItems,
+      existingItem: ids.map(id => existingItemsById[id]),
```

Comment:
There's nothing to guarantee the order of items in `existingItems`, so once we have them we need to map them back into the correct order before handing them off to the hooks.

Reply:
Perhaps worth adding this as a code comment?

Reply:
one doubt, does `getAccessControlledItems` filters out items we have no access? then `existingItem` will be undefined, and there will be undesired update to item if someone is not checking existingItem be undefined. right?